### ISSUE TITLE
I3SinglePandelFitter in lilliput segfaults when GCD file is missing (Trac #2056)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2048]
When using the I3SinglePandelFitter on a file when no GCD file is attached the code is interrupted with a "Segmentation fault" which makes the debugging a lot harder than it should be.

Initially found in: http://code.icecube.wisc.edu/svn/meta-projects/icerec/releases/IC2013-L2_V13-06-00_IceSim4compat_V3/

Confirmed in: http://code.icecube.wisc.edu/svn/meta-projects/icerec/releases/IC2015-L2_V15-04-02/

So I'm assuming it's still there in the most recent versions.

Example script (IceRec 2012 version):

```text
from icecube import dataclasses,dataio,icetray,lilliput
import I3Tray

pulses = 'SRTInIcePulses'

tray = I3Tray.I3Tray()
tray.AddModule("I3Reader",'reader',filenamelist=["/data/sim/IceCube/2012/filtered/level2/neutrino-generator/11029/00000-00999/Level2_nugen_numu_IC86.2012.011029.000000.i3.bz2"])
tray.AddSegment(lilliput.I3SinglePandelFitter, 'test_MPEFit',
            pulses       = pulses,
            domllh       = 'MPE',
            seeds        = [ 'SPEFit2' ],
            trayname     = 'default',
            )
tray.AddModule("Dump","Dump")
tray.AddModule('I3Writer','writer',FileName="./test.i3.bz2",streams=[icetray.I3Frame.DAQ,icetray.I3Frame.Physics])
tray.Execute(10)
tray.Finish()
```

Migrated from https://code.icecube.wisc.edu/ticket/2056
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:14:55", 
    "description": "When using the I3SinglePandelFitter on a file when no GCD file is attached the code is interrupted with a \"Segmentation fault\" which makes the debugging a lot harder than it should be.\n\nInitially found in: http://code.icecube.wisc.edu/svn/meta-projects/icerec/releases/IC2013-L2_V13-06-00_IceSim4compat_V3/\n\nConfirmed in: http://code.icecube.wisc.edu/svn/meta-projects/icerec/releases/IC2015-L2_V15-04-02/\n\nSo I'm assuming it's still there in the most recent versions.\n\nExample script (IceRec 2012 version):\n\n{{{\nfrom icecube import dataclasses,dataio,icetray,lilliput\nimport I3Tray\n\npulses = 'SRTInIcePulses'\n\ntray = I3Tray.I3Tray()\ntray.AddModule(\"I3Reader\",'reader',filenamelist=[\"/data/sim/IceCube/2012/filtered/level2/neutrino-generator/11029/00000-00999/Level2_nugen_numu_IC86.2012.011029.000000.i3.bz2\"])\ntray.AddSegment(lilliput.I3SinglePandelFitter, 'test_MPEFit',\n            pulses       = pulses,\n            domllh       = 'MPE',\n            seeds        = [ 'SPEFit2' ],\n            trayname     = 'default',\n            )\ntray.AddModule(\"Dump\",\"Dump\")\ntray.AddModule('I3Writer','writer',FileName=\"./test.i3.bz2\",streams=[icetray.I3Frame.DAQ,icetray.I3Frame.Physics])\ntray.Execute(10)\ntray.Finish()\n}}}\n\n\n", 
    "reporter": "wvandriessche", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067295757382", 
    "component": "combo reconstruction", 
    "summary": "I3SinglePandelFitter in lilliput segfaults when GCD file is missing", 
    "priority": "normal", 
    "keywords": "lilliput gcd", 
    "time": "2017-07-28T08:58:25", 
    "milestone": "", 
    "owner": "kkrings", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170728T09:04:09: 
* kjmeagher changed milestone from "" to "IceRec 5.2"
* kjmeagher changed owner from "" to "kkrings"
* kjmeagher changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20170728T14:36:49: joertlin commented:

> I also tested it with icerec V05-01-06 and I found the same behaviour.

**Comment by icecube-trac** [comment]
Trac update at 20170728T14:50:22: 
* kkrings changed status from "assigned" to "closed"
* kkrings changed resolution from "" to "fixed"
* kkrings commented:

> In changeset 157415:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="157415"
> Fix potential segmentation fault due to missing geometry; closes #2048.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:14:55: 
* olivas changed milestone from "IceRec 5.2" to ""
* olivas commented:

> Milestone IceRec 5.2 deleted
